public/js/controllers/retrieveVoteCtrl.js

```diff
@@ -0,0 +1,19 @@
+angular.module('voteApp').controller('retrieveVoteController',
+['$scope', 'alertService', 'voteService', 'localStorageService',
+function($scope, alertService, voteService, localStorageService) {
+    $scope.vote = { hash: localStorageService.get('voteHash')};
+
+    $scope.retreiveVote =  function(vote) {
+        voteService.retrieve(vote.hash)
+            .success(function(vote) {
+                var msg =
+                    'Funnet! Stemme med hash "' + vote.hash +
```

Comment:
This should probably be done in the template, and not by adding an alert. Can use `ng-if` to choose if it should be shown or not, and bind the vote to scope (`$scope.vote = vote`).